coalaip_bigchaindb/plugin.py

```diff
@@ -1 +1,114 @@
+
+class Plugin:
+    """BigchainDB ledger plugin for COALA IP's Python reference
+    implementation (LINK).
+
+    Plugs in a BigchainDB instance as the persistence layer for COALA IP
+    related actions.
+    """
+
+    def __init__(self, *nodes):
+        """Initialize a :class:`~coalaip_bigchaindb.Plugin` instance
+        and connect to one or more BigchainDB nodes.
+
+        Args:
+            *nodes (str): one or more URLs of BigchainDB nodes to
+                connect to as the persistence layer
+        """
+
+        self.driver = BigchainDB(*nodes)
+
+    def create_user(self):
+        """Create a new verifying/signing keypair for use with BigchainDB
+
+        Returns:
+            tuple: a new verifying key ('verifying_key') and signing key
+                ('signing_key') pair.
+        """
+
+        return generate_keypair()
+
+    def get_status(self, persist_id):
+        """Get the status of an COALA IP entity on BigchainDB
+
+        Args:
+            persist_id (str): the ID of the creation transaction for the
+                entity on the connected BigchainDB instance
+
+        Returns:
+            Either:
+                str: the status of the entiy; one of:
+                    - 'valid': the transaction has been written in a
+                               validated block
+                    - 'invalid': the block the transaction was in was
+                                 voted invalid
+                    - 'undecided': the block the transaction is in is
+                                   still undecided
+                    - 'backlog': the transaction is still in the backlog
+                None: if 'persist_id' is not a string or no transaction
+                    whose 'uuid' matches 'persist_id' could be found in
+                    the connected BigchainDB instance
+        """
+
+        # FIXME: how to do this return signature better
+
+        # FIXME: maybe just do this in a try/catch block if the driver throws
+        if not isinstance(persist_id, str):
+            return None
+        # FIXME: check API once implemented
+        return self.driver.transactions.get_status(persist_id)
+
+    def save(self, entity_data, *, user):
+        """Create and assign a new entity with the given data to the
+        given user's verifying key on BigchainDB
+
+        Args:
+            entity_data (dict): a dict holding the entity's data that
+                will be saved in a new asset's asset definition
+            user (tuple): a tuple containing:
+                'verifying_key' (str): the user's verifying key on the
+                    connected BigchainDB instance
+                'signing_key' (str): the user's signing key on the
+                    connected BigchainDB instance
+
+        Returns:
+            str: the id of the creation transaction for the new entity
+        """
+
+        # FIXME: how to do the dict arg better
+        # FIXME: should I double check the user dict?
```

Comment:
Two things when I raise errors usually:

- When you want to be nice to the user of a module (because it makes their lifes easier - but maybe also not :D)
- When the error you raise could be recovered.

Since I think there is no recovery from a missing signing or verifying key, and since the already invoked error is quite clear to the outside user, I think you don't have to do an extra check here.

@vrde Would you agree?